specification/resources/resource-manager/Microsoft.Authorization/stable/2019-01-01/policyDefinitions.json

```diff
@@ -0,0 +1,520 @@
+            "schema": {
+              "$ref": "#/definitions/PolicyDefinitionListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Authorization/policyDefinitions": {
+      "get": {
+        "tags": [
+          "PolicyDefinitions"
+        ],
+        "operationId": "PolicyDefinitions_ListBuiltIn",
+        "summary": "Retrieve built-in policy definitions",
+        "description": "This operation retrieves a list of all the built-in policy definitions.",
+        "x-ms-examples": {
+          "List built-in policy definitions": {
+            "$ref": "./examples/listBuiltInPolicyDefinitions.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of built-in policy definitions.",
+            "schema": {
+              "$ref": "#/definitions/PolicyDefinitionListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementgroups/{managementGroupId}/providers/Microsoft.Authorization/policyDefinitions": {
+      "get": {
+        "tags": [
+          "PolicyDefinitions"
+        ],
+        "operationId": "PolicyDefinitions_ListByManagementGroup",
+        "summary": "Retrieve policy definitions in a management group",
+        "description": "This operation retrieves a list of all the policy definitions in a given management group.",
+        "x-ms-examples": {
+          "List policy definitions by management group": {
+            "$ref": "./examples/listPolicyDefinitionsByManagementGroup.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ManagementGroupIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of policy definitions.",
+            "schema": {
+              "$ref": "#/definitions/PolicyDefinitionListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PolicyDefinitionProperties": {
+      "properties": {
+        "policyType": {
+          "type": "string",
+          "description": "The type of policy definition. Possible values are NotSpecified, BuiltIn, and Custom.",
+          "enum": [
+            "NotSpecified",
+            "BuiltIn",
+            "Custom"
+          ],
+          "x-ms-enum": {
+            "name": "policyType",
+            "modelAsString": true
+          }
+        },
+        "mode": {
```

Comment:
With the new 2019-01-01 version, we added data plane policy support. Earlier mode can have only two values - All, Indexed. Now with data plane, it can have more values - one per data plane partner. For example, KeyVault uses Microsoft.KeyVault.Data, AKS uses Microsoft.ContainerService.Data. Removed enum to accommodate data plane partners dynamically.

Reply:
With the new api-version, we ensured existing clients who are using old api-versions do not break when they do list operation.

Reply:
can you have mode = All, indexed, dataplane(ex). and then when its dataplane, you can introduce another property to indicate which data plane provider. 
If mode is unbounded, it will become hard to use. how will customers know which modes are supported.

Reply:
@RamyasreeChakka - thoughts on above?

Reply:
@ravbhatnagar Even with another property case, customer needs to know what all data plane providers are supported. Currently user can know about the available modes through documentation. Data plane policies also will be available here https://docs.microsoft.com/en-us/azure/governance/policy/samples/ 

For discovering data plane aliases, there is a data aliases API.

Reply:
@RamyasreeChakka - but that other property can be an enum to which you can keep adding the supported values. Reading docs to use an SDK/API is not ideal. What is the dataplane aliases API?

Reply:
@ravbhatnagar Adding a value to enum requires a code change, SDK change - it's not dynamic anymore. We want to onboard data plane partners dynamically. Similar to RP onboarding, data plane partners onboard to policy by providing a data manifest and then everything is driven off from manifest. The manifest contains the data plane specific aliases and effects. Currently there is a preview API for get dataplane aliases and for GA planning to add get DataManifests API as well.

REST API to fetch dataAliases looks like below…
GET https://api-dogfood.resources.windows-int.net/providers/microsoft.authorization/dataAliases?api-version=2018-06-01-preview

Reply:
Discussed this with @RamyasreeChakka  and the dataAliases api for GA will include the policyMode property. That will help customers get the value needed for the mode property. FOr knowing the aliases, customers anyway have to call the dataAliases API.